### PR TITLE
runner: refactor to make some steps parallel

### PR DESCRIPTION
**Pull request by Omrigan**
Supersedes #790, part of a #711.

Per-commit review can be done. Only the last commit should have functional changes, the rest is ground work.

Motivation: all the tasks in an errgroup include running external commands, so parallelizing it might yield small latency reduction. 

Didn't test this, is our CI sufficient to check such changes?

### CODE REVIEW

**Review by shayanh — COMMENTED** [review]
Thanks for the PR. Improving runner's codebase significantly. Added two comments. 

Regarding testing, I'd say the CI tests runner to some extent but many configurations remain untouched. Your changes allows us to write unit tests for runner but I'm not sure if we want to take that route right now. To be more safe we can deploy this in two parts (ground work, parallel steps).

File: neonvm/runner/main.go [hunk 1]
```diff
@@ -666,31 +662,55 @@ func run(logger *zap.Logger) error {
...
-	}
 
-	// resize rootDisk image of size specified and new size more than current
-	err = resizeRootDisk(logger, vmSpec)
-	if err != nil {
-		return fmt.Errorf("failed to resize rootDisk: %w", err)
-	}
+	eg := &errgroup.Group{}
+	eg.Go(func() error {
+		err = runInitScript(logger, vmSpec.InitScript)
```
[the remark this answers is not in the document]
Comment (Omrigan) [reply neonvm/runner/main.go]: Thanks! For some reason I remembered about it in the last closure, but forgot in the other 3.

File: neonvm/runner/main.go [hunk 2]
```diff
@@ -666,31 +662,55 @@ func run(logger *zap.Logger) error {
...
+		if err != nil {
+			return fmt.Errorf("failed to run init script: %w", err)
+		}
+		return nil
+	})
+
+	eg.Go(func() error {
+		err = createISO9660runtime(
```
[the remark this answers is not in the document]
Comment (Omrigan) [reply neonvm/runner/main.go]: I usually don't like this nesting, e.g. it is harder to change `fn` arguments in this complex expression. But I've changed it, maybe I'll grow to like it!

**Review by shayanh — APPROVED** [review]

**Review by sharnoff — APPROVED** [review]
looks great, nice!

**Pull request merged**